field/binary_test.go

```diff
@@ -140,3 +141,76 @@ func TestBinaryNil(t *testing.T) {
 	bs = str.Value()
 	require.Nil(t, bs)
 }
+
+func TestBinaryFieldUnmarshal(t *testing.T) {
+	testValue := []byte{0x12, 0x34, 0x56}
+	str := NewBinaryValue(testValue)
+
+	val1 := &Binary{}
+	err := str.Unmarshal(val1)
+	require.NoError(t, err)
+	require.Equal(t, testValue, val1.Value())
+
+	var val2 string
+	err = str.Unmarshal(&val2)
+	require.NoError(t, err)
+	require.Equal(t, "123456", val2)
+
+	var val3 []byte
+	err = str.Unmarshal(&val3)
+	require.NoError(t, err)
+	require.Equal(t, testValue, val3)
+
+	val4 := reflect.ValueOf(&val2).Elem()
+	err = str.Unmarshal(val4)
+	require.NoError(t, err)
+	require.Equal(t, "123456", val4.String())
+
+	val5 := reflect.ValueOf(&val3).Elem()
+	err = str.Unmarshal(val5)
+	require.NoError(t, err)
+	require.Equal(t, testValue, val5.Bytes())
+
+	val6 := reflect.ValueOf(val2)
+	err = str.Unmarshal(val6)
+	require.Error(t, err)
+	require.Equal(t, "cannot set reflect.Value of type string", err.Error())
+
+	val7 := reflect.ValueOf(&val2)
```

Comment:
I would suggest to name variables a little bit differently. In this line, it's not clear what is the type of `val2`.

What about the following changes:

```
testValue := []byte{0x12, 0x34, 0x56} -> expectedBytes
val1 := &Binary{} -> binary or binaryField
var val2 string -> str or s
val3 []byte -> bytes
```

If you agree, we can do this for all tests.

Reply:
agree